### PR TITLE
Add yacc provider and add dependency to swig

### PR DESCRIPTION
**Pull request by AndrewGaspar**

### CODE REVIEW

**Comment by AndrewGaspar** [comment]
cc @sethrj

**Review by sethrj — COMMENTED** [review]
Good implicit dependency catch, but SWIG can build with (and in fact prefers) `bison -y` as well. Since both packages are in Spack but have no clear relation, maybe we need to add a virtual package that both provide. Are you familiar enough with spack to do that?

**Comment by AndrewGaspar** [comment]
Good point. Yeah, I've got it.

File: etc/spack/defaults/packages.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ packages:
       szip: [libszip, libaec]
       tbb: [intel-tbb]
       unwind: [libunwind]
-      sycl: [hipsycl]
```
Comment (AndrewGaspar) [inline etc/spack/defaults/packages.yaml:49]: FYI: This was a duplicate from L45

Comment (sethrj) [reply etc/spack/defaults/packages.yaml:49]: L45?

Comment (sethrj) [reply etc/spack/defaults/packages.yaml:49]: It looks from a `git blame` like this was added unintentionally by https://github.com/spack/spack/pull/14682 and not caught in review. Grepping through the spack packages doesn't show any `depends_on('sycl')` so we can let this change through and see if anything bad happens?

**Review by sethrj — APPROVED** [review]
Aside from the comment about `sycl` deletion (which isn't quite relevant to this commit, nor was it to the PR that introduced it), it looks good.

File: etc/spack/defaults/packages.yaml (hunk 1 above)
Comment (AndrewGaspar) [reply etc/spack/defaults/packages.yaml:49]: line 45. just up the page - it was a duplicated definition.

Comment (AndrewGaspar) [reply etc/spack/defaults/packages.yaml:49]: https://github.com/spack/spack/pull/19087/files/d5510ed8c31d2590ac17560fc94199205cc12b9d#diff-50db37ac9e31f0e8d94666e865343cb0R45

**Pull request merged**